binding.cc

```diff
@@ -442,19 +446,18 @@ namespace zmq {
 
         // get our next frame it may have the target address and safely copy to our buffer
         zmq_msg_init (&msg2);
-        if (zmq_msg_more(&msg1) == 0) {
+        if (zmq_msg_more(&msg1) == 0 || zmq_recvmsg (s->monitor_socket_, &msg2, 0) == -1) {
           NanThrowError(ExceptionFromError());
-          return;
-        }
-        if (zmq_recvmsg (s->monitor_socket_, &msg2, 0) == -1) {
-          NanThrowError(ExceptionFromError());
-          return;
+          zmq_msg_close(&msg2);
+          continue;
```

Comment:
My intention on that check was to fail and stop everything if we did not have a second message. 
This would indicate a serious underlying problem if we do not receive both.
 
zmq::socket_base_t::monitor_event sends both messages and since they are not true sockets but internal messages the only reason we do not get both would be a serious failure inside of zmq.  I never bother to close msg2 my thinking was that at this point everything was fubar anyway. 

I needed to better comment that if() to record my thoughts. I had no docs or could not find any docs on handing an exception inside of a timer event and how it would react to NanThrowError I made an assumption it would expel a message someplace inside of native js code? I agree with @kkoopa that this was what I wanted to have happen in this case and again be able to be trapped in native js code someplace if needed to do some total restart in this case.